### ISSUE TITLE
SDK Crash when ending the call / navigating away

### ISSUE THREAD

**Issue by ac-opensource** [livekit/components-android#16]
**Describe the bug**
SDK Crashes the implementing app when navigating away from the screen that implements RoomScope

**To Reproduce**
1. Implement [https://github.com/livekit/components-android?tab=readme-ov-file#basic-usage](https://github.com/livekit/components-android?tab=readme-ov-file#basic-usage)
2. Navigate Up via Compose Navigation

**Expected behavior**
The app should go back to previous screen normally

**Screenshots**
The app crashes

**Device Info:**
 - Device: Emulator, Huawei Mate 50 pro, Xiaomi 11T 
 - OS: Vanilla Ice Cream, Harmony OS 4.2, Android 12
 - LiveKit SDK version: 1.1.1 

**Additional context**

Crash Logs:

```
2024-06-03 18:33:29.861 13578-13613 EGL_emulation           com.example.app                 D  app_time_stats: avg=68.45ms min=13.09ms max=109.13ms count=16
2024-06-03 18:33:30.052 13578-13779 EGL_emulation           com.example.app                 D  app_time_stats: avg=88.14ms min=65.25ms max=116.72ms count=13
2024-06-03 18:33:30.786 13578-13779 libEGL                  com.example.app                 E  call to OpenGL ES API with no current context (logged once per thread)
2024-06-03 18:33:30.789 13578-13784 libEGL                  com.example.app                 E  call to OpenGL ES API with no current context (logged once per thread)
2024-06-03 18:33:30.813 13578-13792 libc                    com.example.app                 A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x4 in tid 13792 (EncoderQueue - ), pid 13578 (com.example.app)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A  Cmdline: com.example.app
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A  pid: 13578, tid: 13792, name: EncoderQueue -  >>> com.example.app <<<
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #00 pc 0000000000391a44  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #01 pc 00000000008c2184  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #02 pc 00000000008c2200  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #03 pc 00000000008d0f50  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #04 pc 00000000008d12c4  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #05 pc 0000000000620a6c  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #06 pc 000000000061fec8  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #07 pc 00000000006da048  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #08 pc 00000000006da048  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #09 pc 00000000008d06f0  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #10 pc 00000000008cfc44  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #11 pc 00000000008cf224  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #12 pc 00000000008c4194  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #13 pc 000000000080b494  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #14 pc 000000000080d044  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #15 pc 000000000080bc98  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #16 pc 0000000000440a00  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:33:31.120 13799-13799 DEBUG                   pid-13799                            A        #17 pc 00000000004408e8  /data/app/~~vtD81IKFPIkE7_qh8cDmNQ==/com.example.app-RTE3MOhcGKBjfSP_KbynbQ==/lib/arm64/liblkjingle_peerconnection_so.so (BuildId: 70ca83c0b27c0940)
2024-06-03 18:39:52.045 13845-13845 FirebaseSessions        com.example.app                 D  Initializing Firebase Sessions SDK.
2024-06-03 18:39:52.047 13845-13845 FirebaseCrashlytics     com.example.app                 I  Initializing Firebase Crashlytics 19.0.0 for com.example.app
2024-06-03 18:39:52.052 13845-13845 SessionsDependencies    com.example.app                 D  Subscriber CRASHLYTICS registered.
```

**Comment by alexsullivan-substack** [comment]
Fwiw downgrading to 1.1.0 seems to have fixed this issue for me.

**Comment by davidliu** [comment]
@ac-opensource @alexsullivan-substack Fixed in 1.1.2 release.
